### PR TITLE
Fix/liger fused linear cross entropy function does not support reduction=none

### LINKED ISSUE #488: LigerFusedLinearCrossEntropyFunction does not support reduction=None
### 🐛 Describe the bug

loss = LigerFusedLinearCrossEntropyLoss(reduction='none')(model.lm_head.weight, flattened_hidden_states, flattened_target)

returns loss with shape []:
tensor(209594.4062, device='cuda:7',
       grad_fn=<LigerFusedLinearCrossEntropyFunctionBackward>)
       
reduction is actually perfromed

### Reproduce

```python
from liger_kernel.transformers import LigerCrossEntropyLoss, LigerFusedLinearCrossEntropyLoss
device3 = 'cuda'
weight = torch.randn((180000, 4096), device=device3, dtype=torch.float32)
fhidden_states = torch.randn((20, 4096), device=device3, dtype=torch.float32)
ftarget = torch.ones((20,), device=device3, dtype=torch.long)
loss = LigerFusedLinearCrossEntropyLoss(reduction='none')(weight, fhidden_states, ftarget)
print(loss)
```

### Versions

Environment Report:
-------------------
Operating System: Linux-5.4.0-135-generic-x86_64-with-glibc2.31
Python version: 3.10.14
Liger Kernel version: 0.5.2
PyTorch version: 2.5.1+cu124
CUDA version: 12.4
HIP(ROCm) version: Not available
Triton version: 3.1.0
Transformers version: 4.44.0
XPU version: XPU Not Available

### PR DESCRIPTION
**Pull request by ryankert01**
## Summary
fix #488 

## Testing Done

- Hardware Type: A100-40G-PCIe
- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

### CODE REVIEW

**Comment by Tcc0403** [comment]
LGTM

**Review by Tcc0403 — APPROVED** [review]
LGTM

**Review by Tcc0403 — APPROVED** [review]

**Comment by Tcc0403** [comment]
All tests passed on my machine.
